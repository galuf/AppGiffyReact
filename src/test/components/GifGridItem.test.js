import GridGifItem from "../../components/GifGridItem";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

describe("Test de <GifgridItem />", () => {
  const title = "Un titulo";
  const url = "https://localhost/algo.jpg";
  const wrapper = shallow(<GridGifItem title={title} url={url} />);

  test("Debe mostrar el componente correctamente ", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test("Debe de tener un parrafo con el title ", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("Debe de tener la imagen igual al url y alt de los props", () => {
    const img = wrapper.find("img");
    //console.log(img.prop("src"));
    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
  });

  test("Debe tener animate__fadeIn ", () => {
    const div = wrapper.find("div.card");
    //console.log(div.hasClass("animate__fadeIn"));
    expect(div.hasClass("animate__fadeIn")).toBe(true);
  });
});
