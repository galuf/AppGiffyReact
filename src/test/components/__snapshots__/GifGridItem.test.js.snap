// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de <GifgridItem /> Debe mostrar el componente correctamente  1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un titulo"
    src="https://localhost/algo.jpg"
  />
  <div>
    <p>
      Un titulo
    </p>
  </div>
</div>
`;
